airbyte/cloud/_connections.py

```diff
@@ -0,0 +1,206 @@
+        return [stream.name for stream in self._connection_info.configurations.streams]
+
+    @property
+    def table_prefix(self) -> str:
+        """The table prefix."""
+        if not self._connection_info:
+            self._connection_info = self._fetch_connection_info()
+
+        return self._connection_info.configurations.prefix
+
+    @property
+    def connection_url(self) -> str | None:
+        return f"{self.workspace.workspace_url}/connections/{self.connection_id}"
+
+    @property
+    def job_history_url(self) -> str | None:
+        return f"{self.connection_url}/job-history"
+
+    # Run Sync
+
+    def run_sync(
+        self,
+        *,
+        wait: bool = True,
+        wait_timeout: int = 300,
+    ) -> SyncResult:
+        """Run a sync."""
+        connection_response = api_util.run_connection(
+            connection_id=self.connection_id,
+            api_root=self.workspace.api_root,
+            api_key=self.workspace.api_key,
+            workspace_id=self.workspace.workspace_id,
+        )
+        sync_result = SyncResult(
+            workspace=self.workspace,
+            connection=self,
+            job_id=connection_response.job_id,
+        )
+
+        if wait:
+            sync_result.wait_for_completion(
+                wait_timeout=wait_timeout,
+                raise_failure=True,
+                raise_timeout=True,
+            )
+
+        return sync_result
+
+    # Logs
+
+    def get_previous_sync_logs(
+        self,
+        *,
+        limit: int = 10,
+    ) -> list[SyncResult]:
+        """Get the previous sync logs for a connection."""
+        sync_logs: list[JobResponse] = api_util.get_job_logs(
+            connection_id=self.connection_id,
+            api_root=self.workspace.api_root,
+            api_key=self.workspace.api_key,
+            workspace_id=self.workspace.workspace_id,
+            limit=limit,
+        )
+        return [
+            SyncResult(
+                workspace=self.workspace,
+                connection=self,
+                job_id=sync_log.job_id,
+                _latest_status=sync_log.status,
+            )
+            for sync_log in sync_logs
+        ]
+
+    def get_sync_result(
+        self,
+        job_id: str | None = None,
+    ) -> SyncResult | None:
+        """Get the sync result for the connection.
+
+        If `job_id` is not provided, the most recent sync job will be used.
+
+        Returns `None` if job_id is omitted and no previous jobs are found.
+        """
+        if job_id is None:
+            # Get the most recent sync job
+            results = self.get_previous_sync_logs(
+                limit=1,
+            )
+            if results:
+                return results[0]
+
+            return None
+
+        # Get the sync job by ID (lazy loaded)
+        return SyncResult(
+            workspace=self.workspace,
+            connection=self,
+            job_id=job_id,
+        )
+
+    # Deletions
+
+    def delete(
```

Comment:
Are we making deletion too easy? On the UI, when I try to delete a connection, I get the following message:
Deleting a connection cannot be undone without a full re-sync. Note that:
- All past logs and configurations will be deleted
- Updates of new data will stop
- No existing data in the destination will be altered
Type GitHub → Google Sheets below to confirm this action.

Reply:
I think I agree...

I though that the same when designing these. I will think a bit more on this... 🤔

If we aren't sure about the API as of now, we could make private members (`_delete()`); we could also try to phase more scarily `_delete_forever()` - or maybe force the user to confirm by passing the connection ID back to itself or turning on a 'mode' that allows destructive actions.

Reply:
yea I like the idea of making it private. We could do the same with the delete_connection method in deploy context?

Reply:
Update: `deploy*()` methods have now all been renamed to `_deploy*()`.